types/api/v1beta1/event.proto

```diff
@@ -0,0 +1,162 @@
+    repeated string actions = 2;
+}
+
+message Context {
+    Process process = 1;
+    optional Container container = 2;
+    optional Kubernetes kubernetes = 3;
+}
+
+message Process  {
+    int64 executionTime = 1;
+    Executable executable = 2;
+    int64 pid = 3;
+    int64 namespaced_pid = 4;
+    uint32 entity_id = 5;
+    User real_user = 6;
+    User user = 7;
+    Parent parent = 8;
+    Thread thread = 9;       
+}
+
+message Executable {
+    string name = 1;
+    string path = 2;
+}
+
+message User {
+    int64 id = 1;
+    string name = 2;
+}
+
+message Parent {
+    int64 pid = 1;
+    uint32 entityId = 2;
+    Executable executable = 3;
+    int64 start = 4;
+    optional Parent parent = 5;
+}
+
+message Thread {
+    int64 start = 1;
+    string name = 2;
+    int64 tid = 3;
+    int64 namespaced_tid = 4;
+    repeated string capabilities = 5;
+    string syscall = 6;
+    bool compat = 7;
+    repeated uint64 user_stack_strace = 8;
+}
+
+message Container {
+    string id = 1;
+    string name = 2; 
+    ContainerImage image = 3;
+    bool started = 4;
+    int64 start = 5;
+    int64 pid = 6;
+}
+
+message ContainerImage {
+    string id = 1;
+    string repo_digest = 2;
+    string name = 3;
+}
+
+message Kubernetes {
+    Pod pod = 1;
+    KubernetesNamespace namespace = 2;
+}
+
+message Pod {
+    string name = 1;
+    string uid = 2;
+    map<string, string> labels = 3;
+}
+
+message KubernetesNamespace {
+    string name = 1;
+}
+
+message Threat {
+    string description = 1;
+    MitreTatic mitre_tactic = 2;
+    MitreTechinique mitre_technique = 3;
+    string severity = 4;
+}
+
+message MitreTatic {
+    string name = 1;
+}
+
+message MitreTechinique {
+    string id = 1;
+    string external_id = 2;
+    string name = 3;
+}
+
+// TODO:  what do you think about event data like this? Please read below the reasoning
+// document the Value message.
+message EventData {
+  string name = 1;
+  ValueType valueType = 2;
+  Value value = 3;
+}
+
+// ValueType enumerates types of Argument values.
+enum ValueType {
+  UNKNOWN_VALUE_TYPE = 0;
+  STRING = 1;
+  STRING_ARRAY = 2;
+  UINT32 = 3;
+  UINT64 = 4;
+  INT32 = 5;
+  INT64 = 6;
+  TRIGGERING_EVENT = 7;
+}
+
+// Value is a union-like structure that wraps the value of a given Argument. We
+// are not using Oneof nor Any type because of their limitations. The Oneof type
+// does not allow repeated values, whereas Any is not convenient to unpack
+// structured values.
+//
+// A similar approach is taken by Kubernetes APIs. For example,
+// PersistentVolumeSource holds mutually exclusive volume definitions.
+// See https://github.com/kubernetes/api/blob/master/core/v1/generated.proto
+//
+message Value {
+  optional string stringValue = 1;
+  repeated string stringArrayValue = 2;
+  optional uint32 uint32Value = 3;
+  optional uint64 uint64Value = 4;
+  optional int32 int32Value = 5;
+  optional int64 int64Value = 6;
+  optional TriggeringEvent triggering_event = 7;
+}
```

Comment:
Questions:
1. What if there is more than one value of a specific type? E.g. two strings. How is that supported?
2. since stringArrayValue is repeated - does it mean it is not optional? in other words, stringArrayValue will always appear, no?
3. There are other types not included here - are you going to add them as well?